<launch>

    <arg name="ns_prefix" default="uav_"/>
    <arg name="robot_model" default="iris"/>
    <arg name="pub_rate" default="50.0"/>
    <arg name="debug" default="false"/>
    <arg name="save_data" default="false"/>
    <arg name="trajectory" default="true"/>
    <arg name="vxy" default="2.0"/>
    <arg name="vz_up" default="3.0"/>
    <arg name="vz_dn" default="1.0"/>
    <arg name="reach_tolerance" default="0.1"/>
    <arg name="path" default="cubic"/>
    <arg name="use_class" default="true" unless="$(arg debug)"/>
    <arg name="use_class" default="false" if="$(arg debug)"/>
    <arg name="multi" default="true" unless="$(arg debug)"/>
    <arg name="multi" default="false" if="$(arg debug)"/>

    <group ns="$(arg ns_prefix)1">
        <node pkg="upat_follower" type="ual_communication_node" name="ual_communication" output="screen" required="true">
            <param name="uav_id" value="1"/>
            <param name="save_csv" value="$(arg save_data)"/>
            <param name="trajectory" value="$(arg trajectory)"/>
            <param name="pub_rate" value="$(arg pub_rate)"/>
            <param name="path" value="$(arg path)"/>
            <param name="reach_tolerance" value="$(arg reach_tolerance)"/>
            <param name="use_class" value="$(arg use_class)"/>
        </node>
        <node pkg="upat_follower" type="follower_node" name="follower" output="screen" required="true" unless="$(arg use_class)">
            <param name="uav_id" value="1"/>
            <param name="debug" value="$(arg debug)"/>
            <param name="pub_rate" value="$(arg pub_rate)"/>
            <param name="vxy" value="$(arg vxy)"/>
            <param name="vz_up" value="$(arg vz_up)"/>
            <param name="vz_dn" value="$(arg vz_dn)"/>
        </node>
        <node pkg="upat_follower" type="visualization_node" name="visualization" required="true">
            <param name="uav_id" value="1"/>
            <param name="robot_model" value="$(arg robot_model)"/>
            <param name="pub_rate" value="$(arg pub_rate)"/>
        </node>
</launch>